expression/expression.go

```diff
@@ -126,10 +138,34 @@ func (s Schema) FindColumn(astCol *ast.ColumnName) (*Column, error) {
 		if (dbName.L == "" || dbName.L == col.DBName.L) &&
 			(tblName.L == "" || tblName.L == col.TblName.L) &&
 			(colName.L == col.ColName.L) {
-			if idx != -1 {
-				return nil, errors.Errorf("Column '%s' is ambiguous", colName.L)
+			if idx == -1 {
+				idx = i
+			} else {
+				return nil, errors.Errorf("Column %s is ambiguous", col.ToString())
+			}
+		}
+	}
+	if idx == -1 {
+		return nil, nil
+	}
+	return s[idx], nil
+}
+
+// FindSelectFieldColumn finds a column from select fields.
+func (s Schema) FindSelectFieldColumn(astCol *ast.ColumnName, selectFields []Expression) (*Column, error) {
+	dbName, tblName, colName := astCol.Schema, astCol.Table, astCol.Name
+	idx := -1
+	for i, col := range s {
+		if (dbName.L == "" || dbName.L == col.DBName.L) &&
+			(tblName.L == "" || tblName.L == col.TblName.L) &&
+			(colName.L == col.ColName.L) {
+			if expr, ok := selectFields[i].(*Column); !ok {
+				return s[i], nil
```

Comment:
It's wrong, because having or order by clause need their child projection's output columns, but not input expressions.